### DISCUSSION TITLE
Periodic syncing is Unhealthy problem

### DISCUSSION THREAD

**Discussion by the-mace** [philosowaffle/peloton-to-garmin#496] [Q&A] [answered]
Hi folks, I have things setup in Docker with the webui. In the webui, syncing is working fine on manual triggers and I see the results in Garmin connect. So all that is good.

The periodic syncing says "Periodic syncing is Unhealthy" in the UI and in the logs I get:
```
2023-05-17 09:56:17.038 -05:00 [INF] Sleeping for 60 seconds...
2023-05-17 09:57:18.249 -05:00 [INF] Found 5 completed workouts.
2023-05-17 09:57:19.928 -05:00 [INF] Found 0 workouts remaining after filtering ExcludedWorkoutTypes.
2023-05-17 09:57:19.932 -05:00 [INF] Converting workouts...
2023-05-17 09:57:19.935 -05:00 [INF] All converters were skipped. Ensure you have atleast one output Format configured in your settings. Converting to FIT or TCX is required prior to uploading to Garmin Connect.
2023-05-17 09:57:19.966 -05:00 [INF] Sleeping for 60 seconds...
```
I have it syncing often as I try things.

My configuration.local.json looks like this:
```
{
  "Api": {
    "HostUrl": "http://*:8080"
  },
  "Format": {
    "Fit": true
  },
  "Observability": {

    "Prometheus": {
      "Enabled": false
    },

    "Jaeger": {
      "Enabled": false,
      "AgentHost": "localhost",
      "AgentPort": 6831
    },

    "Serilog": {
      "Using": [ "Serilog.Sinks.Console", "Serilog.Sinks.File" ],
      "MinimumLevel": {
        "Default": "Information",
        "Override": {
          "Microsoft": "Error",
          "System": "Error"
        }
      },
      "WriteTo": [
        { "Name": "Console" },
        {
          "Name": "File",
          "Args": {
            "path": "./output/log.txt",
            "rollingInterval": "Day",
            "retainedFileCountLimit": 7
          }
        }
      ]
    }
  }
}
```
This is just the default config with me adding the Format section to try that. 

in /app/data/SettingsDb.json is also the output format as well as my logins for Peloton and Garmin and as I said it syncs manually fine. 

I'm not sure where to go from here. The webui is configured/working fine but the regular sync is failing. Any ideas where to look? Im not doing anything fancy.

**Comment by philosowaffle** [comment]
This error is misleading currently. The root of the problem is that 0 workouts were converted, and in this scenario the code is assuming that this means the converters were skipped due to misconfiguration. In reality they were skipped because there were no workouts to convert. Just pushed a fix to this onto main branch.

**Reply by the-mace** [reply]
Thank you. It's not just that I got an error message though. New workouts didn't sync until I did then manually. Will try again with the latest code.

**Comment by the-mace** [comment]
Ok so I set periodic sync to 600 (10 mins) via UI, then did a `docker-compose down` and then a `docker-compose pull && docker-compose up -d` then did my am workout.

It did not sync to Garmin.

I see this:
![Screenshot 2023-05-22 at 9 21 35 AM](https://github.com/philosowaffle/peloton-to-garmin/assets/2512112/f72cd14e-7e08-452e-b6d0-64fc4df03b45)

![Screenshot 2023-05-22 at 9 21 30 AM](https://github.com/philosowaffle/peloton-to-garmin/assets/2512112/f4947c5e-a7f7-4ec4-8faa-d047cde9de4e)

So it saw the workouts but did not send them to Garmin. Any ideas?

**Comment by philosowaffle** [comment, marked as answer]
The next thing to look at is from your original log message:

```
 Found 5 completed workouts.
Found 0 workouts remaining after filtering ExcludedWorkoutTypes.
```

This indicates you have some workout types configured that P2G should skip syncing by default. (more info in the [docs](https://philosowaffle.github.io/peloton-to-garmin/configuration/json.html#peloton-config)).

On the Web UI in the Peloton Settings section, there is a setting for `Excluded Workout Types`, check if you have any workout types selected in there.

**Comment by the-mace** [comment]
Doh, that was it. I must have misread it and thought it was "included workout types". It seems that setting only applies to periodic sync and not the manual syncs which threw me off. Its working now! Thank you.
